providers/ms_entra_id.go

```diff
@@ -0,0 +1,166 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/util"
+	"github.com/spf13/cast"
+)
+
+// MicrosoftEntraIDProvider represents provider for Azure Entra Authentication V2 endpoint
+type MicrosoftEntraIDProvider struct {
+	*OIDCProvider
+	skipGraphGroups           bool
+	multiTenantAllowedTenants []string
+
+	microsoftGraphURL *url.URL
+	isMultiTenant     bool
```

Comment:
Again is this boolean necessary because the list of allowed tenants is optional?

Reply:
I have run some tests and this is actually indeed redundant and I will remove it.
List of allowed tenants only makes sense for multi-tenants app, however it can serve as an redundant check for single-tenant apps too. I have also simplified the flag names.